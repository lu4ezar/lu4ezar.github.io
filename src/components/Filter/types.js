// @flow
import type { Filter, FilterName, ActionPayload } from "../../types/filter";

export type Props = {|
  filter: Filter,
  handleChange: (filter: FilterName) => void,
  completedCount: number,
  expiredCount: number,
  setFilter: (payload: ActionPayload) => void,
|};
