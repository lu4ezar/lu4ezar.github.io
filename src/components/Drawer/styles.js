import styled from "styled-components";
import { ListColor } from "../../utils/color";

const Content = styled.div`
  display: flex;
  justify-content: ${(props) =>
    props.side === "left" ? "flex-end" : "flex-start"};
  background-color: ${ListColor.background};
`;

export default Content;
