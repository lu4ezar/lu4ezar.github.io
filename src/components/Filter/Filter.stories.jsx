
import React from 'react';
import Filter from './Filter';

export default {
  component: Filter,
  title: 'Filter',
};

const Template = args => <Filter {...args} />;

export const Default = Template.bind({});
Default.args = {
  setFilter,                                           completedCount,
    expiredCount,                                        filter: {
      master: { status: masterFilterStatus },              priority: { status: priorityFilterStatus, value: priorityFilterValue },                                   completed: { status: completedFilterStatus, val
ue: completedFilterValue },                                expired: { status: expiredFilterStatus, value: expiredFilterValue }
    }
};

export const Enabled = Template.bind({});
Enabled.args = {};

