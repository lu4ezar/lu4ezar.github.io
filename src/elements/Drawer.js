// @flow
import * as React from 'react';
import CoreDrawer from '@material-ui/core/Drawer';
import { IconButton, Divider } from '@material-ui/core';
import { ListColor } from '../utils/color';
import { ChevronLeft, ChevronRight } from '@material-ui/icons';

type Props = {
	side: string,
	open: boolean,
	toggleDrawer: () => void,
	variant?: string,
	children: React.Node
};

const Drawer = ({
	side,
	open,
	toggleDrawer,
	children,
	variant
}: Props): React.Node => {
	const Chevron = side === 'left' ? <ChevronRight /> : <ChevronLeft />;
	return (
		<CoreDrawer
			anchor={side}
			open={open}
			onClose={toggleDrawer}
			variant={variant}
			children={children}
		>
			<div
				style={{
					display: 'flex',
					justifyContent: `${
						side === 'left' ? 'flex-end' : 'flex-start'
					}`,
					background: `${ListColor.background}`
				}}
			>
				<IconButton onClick={toggleDrawer}>{Chevron}</IconButton>
			</div>
			<Divider />
			{children}
		</CoreDrawer>
	);
};

export default Drawer;
