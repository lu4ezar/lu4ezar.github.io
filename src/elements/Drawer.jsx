// @flow
import * as React from "react";
import CoreDrawer from "@material-ui/core/Drawer";
import { IconButton, Divider, Container } from "@material-ui/core";
import { ChevronLeft, ChevronRight } from "@material-ui/icons";
import { ListColor } from "../utils/color";

type Props = {
  side: string,
  open: boolean,
  toggleDrawer: () => void,
  children: React.Node,
};

const Drawer = ({ side, open, toggleDrawer, children }: Props): React.Node => {
  const Chevron = side === "left" ? <ChevronRight /> : <ChevronLeft />;
  return (
    <CoreDrawer anchor={side} open={open} onClose={toggleDrawer}>
      <div
        style={{
          display: "flex",
          justifyContent: `${side === "left" ? "flex-end" : "flex-start"}`,
          background: `${ListColor.background}`,
        }}
      >
        <IconButton onClick={toggleDrawer}>{Chevron}</IconButton>
      </div>
      <Divider />
      <Container maxWidth={false}>{children}</Container>
    </CoreDrawer>
  );
};

export default Drawer;
