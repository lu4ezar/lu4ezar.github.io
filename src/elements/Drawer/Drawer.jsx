// @flow
import * as React from "react";
import {
  IconButton,
  Divider,
  Container,
  Drawer as CoreDrawer,
} from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";
import Content from './styles';
import type { Props } from './types';

const Drawer = ({ side, open, toggleDrawer, children }: Props): React.Node => (
  <CoreDrawer anchor={side} open={open} onClose={toggleDrawer}>
    <Content side={side}>
      <IconButton onClick={toggleDrawer}>
        <CloseIcon />
      </IconButton>
    </Content>
    <Divider />
    <Container maxWidth={false}>{children}</Container>
  </CoreDrawer>
);

export default Drawer;
