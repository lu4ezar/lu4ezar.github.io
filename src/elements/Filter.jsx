// @flow
import * as React from "react";
import {
  Grid,
  Switch,
  Select,
  Badge,
  MenuItem,
  FormControlLabel,
  Collapse,
  Paper,
  Typography,
} from "@material-ui/core";
import type {
  Filter as FilterType,
  FilterTitle,
  FilterValue,
  ActionPayload,
} from "../types/filter";

type Props = {
  filter: FilterType,
  completedCount: number,
  expiredCount: number,
  setFilter: (payload: ActionPayload) => void,
};

const Filter = (props: Props) => {
  const {
    setFilter,
    completedCount,
    expiredCount,
    filter: {
      master: { status: masterFilterStatus },
      priority: { status: priorityFilterStatus, value: priorityFilterValue },
      completed: { status: completedFilterStatus, value: completedFilterValue },
      expired: { status: expiredFilterStatus, value: expiredFilterValue },
    },
  } = props;

  const handleChange = (filter: FilterTitle) => (
    event: SyntheticInputEvent<HTMLInputElement>
  ) => {
    const target = event.target.type ? event.currentTarget : event.target; // MUI Select does not have currentTarget
    const value: FilterValue =
      target.type === "checkbox" ? target.checked : target.value;
    const { name } = target;
    setFilter({
      filter,
      property: name,
      value,
    });
  };

  return (
    <Grid container direction="column">
      {/* master filter */}
      <Grid item>
        <FormControlLabel
          label="Filter"
          labelPlacement="top"
          control={
            <Switch
              name="status"
              checked={masterFilterStatus}
              onChange={handleChange("master")}
              value={masterFilterStatus}
            />
          }
        />
      </Grid>
      <Collapse in={masterFilterStatus}>
        <Grid container spacing={4} justify="center">
          {/* priority filter */}
          <Grid item>
            <Paper>
              <Grid container alignItems="center" justify="center">
                <Grid item>
                  <Typography>off</Typography>
                </Grid>
                <Grid item>
                  <Switch
                    name="status"
                    checked={priorityFilterStatus}
                    onChange={handleChange("priority")}
                    value={priorityFilterStatus}
                  />
                </Grid>
                <Grid item>
                  <Typography>on</Typography>
                </Grid>
              </Grid>
              <Grid container alignItems="center" justify="center">
                <Grid item>
                  <Typography>Priority</Typography>
                </Grid>
              </Grid>
              <Grid item>
                <Select
                  name="value"
                  value={priorityFilterValue}
                  onChange={handleChange("priority")}
                  disabled={!masterFilterStatus || !priorityFilterStatus}
                  variant="standard"
                  multiple
                >
                  <MenuItem value="low">low</MenuItem>
                  <MenuItem value="normal">normal</MenuItem>
                  <MenuItem value="high">high</MenuItem>
                </Select>
              </Grid>
            </Paper>
          </Grid>
          {/* completed filter */}
          <Grid item>
            <Paper>
              <Grid container alignItems="center" justify="center">
                <Grid item>
                  <Typography>off</Typography>
                </Grid>
                <Grid item>
                  <Switch
                    name="status"
                    checked={completedFilterStatus}
                    onChange={handleChange("completed")}
                    value="completedFilterStatus"
                  />
                </Grid>
                <Grid item>
                  <Typography>on</Typography>
                </Grid>
              </Grid>
              <Grid container alignItems="center" justify="center">
                <Grid item>
                  <Badge color="primary" badgeContent={completedCount}>
                    <Typography>Completed</Typography>
                  </Badge>
                </Grid>
              </Grid>
              <Grid container alignItems="center">
                <Grid item>
                  <Typography>hide</Typography>
                </Grid>
                <Grid item>
                  <Switch
                    name="value"
                    checked={completedFilterValue}
                    onChange={handleChange("completed")}
                    value="completedFilter"
                    disabled={!masterFilterStatus || !completedFilterStatus}
                  />
                </Grid>
                <Grid item>
                  <Typography>show</Typography>
                </Grid>
              </Grid>
            </Paper>
          </Grid>
          {/* expired filter */}
          <Grid item>
            <Paper>
              <Grid container alignItems="center" justify="center">
                <Grid item>
                  <Typography>off</Typography>
                </Grid>
                <Grid item>
                  <Switch
                    name="status"
                    checked={expiredFilterStatus}
                    onChange={handleChange("expired")}
                    value="expiredFilterStatus"
                  />
                </Grid>
                <Grid item>
                  <Typography>on</Typography>
                </Grid>
              </Grid>
              <Grid container alignItems="center" justify="center">
                <Grid item>
                  <Badge color="secondary" badgeContent={expiredCount}>
                    <Typography>Expired</Typography>
                  </Badge>
                </Grid>
              </Grid>
              <Grid container alignItems="center">
                <Grid item>
                  <Typography>hide</Typography>
                </Grid>
                <Grid item>
                  <Switch
                    name="value"
                    checked={expiredFilterValue}
                    onChange={handleChange("expired")}
                    value="expiredFilter"
                    disabled={!masterFilterStatus || !expiredFilterStatus}
                  />
                </Grid>
                <Grid item>
                  <Typography>show</Typography>
                </Grid>
              </Grid>
            </Paper>
          </Grid>
        </Grid>
      </Collapse>
    </Grid>
  );
};

export default Filter;
