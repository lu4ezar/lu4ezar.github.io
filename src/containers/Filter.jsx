// @flow
import { useApolloClient } from "@apollo/client";
import * as React from "react";
import { filterVar } from "../apollo/cache";
import { GET_TODOS } from "../apollo/queries";
import Filter from "../components/Filter";

export default () => {
  const client = useApolloClient();
  const todos = client.readQuery(GET_TODOS);
  const completedCount = todos.filter((todo) => !!todo.completed).length;
  const expiredCount = todos.filter((todo) => todo.expires > Date.now()).length;
  const filter = filterVar();

  const handleChange = (filterName: FilterName) => (
    event: SyntheticInputEvent<HTMLInputElement>
  ) => {
    // MUI Select does not have currentTarget
    const target: HTMLInputElement = event.target.type
      ? event.currentTarget
      : event.target;
    const value: FilterValue =
      target.type === "checkbox"
        ? (target.checked:
            | FilterStatus
            | CompletedFilterValue
            | ExpiredFilterValue)
        : // $FlowFixMe
          (target.value: PriorityFilterValue);
    const { name } = target;

    filterVar({
      filter: filterName,
      // $FlowFixMe
      property: name,
      value,
    });
  };
  return (
    <Filter
      filter={filter}
      handleChange={handleChange}
      completedCount={completedCount}
      expiredCount={expiredCount}
    />
  );
};
