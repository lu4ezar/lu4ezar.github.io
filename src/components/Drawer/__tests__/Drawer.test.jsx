/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Drawer from "../Drawer";
import Filter from "../../Filter";

const props = {
  side: "left",
  open: true,
  toggleDrawer: jest.fn(),
  children: <Filter />,
};

jest.mock("../../Filter", () => ({
    __esModule: true,
    default: () => <div />,
  }));

describe("Drawer", () => {
  it("renders as expected", () => {
    const { container } = render(<Drawer {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("renders Close button which calls toggleDrawer function", () => {
    render(<Drawer {...props} />);
    userEvent.click(screen.getByTitle(/close/i));
    expect(props.toggleDrawer).toBeCalled();
  });
});
