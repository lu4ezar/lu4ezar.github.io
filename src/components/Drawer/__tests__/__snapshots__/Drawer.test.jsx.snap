// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders as expected 1`] = `
<div
  aria-hidden="true"
/>
`;
